std/range/package.d

```diff
@@ -8636,65 +9187,68 @@ public:
-
-        // saveable
-        auto s = r.slide(2);
-        assert(s[0 .. 2].equal!equal([[0, 1], [1, 2]]));
-        s.save.popFront;
-        assert(s[0 .. 2].equal!equal([[0, 1], [1, 2]]));
-
-        assert(r.slide(3)[1 .. 3].equal!equal([[1, 2, 3], [2, 3, 4]]));
-    }
-
     alias SliceableDummyRangesWithoutInfinity = AliasSeq!(
         DummyRange!(ReturnBy.Reference, Length.Yes, RangeType.Random, T),
         DummyRange!(ReturnBy.Value, Length.Yes, RangeType.Random, T),
         SliceableRange!(T, No.withOpDollar, No.withInfiniteness),
         SliceableRange!(T, Yes.withOpDollar, No.withInfiniteness),
     );
 
-    foreach (Range; AliasSeq!SliceableDummyRangesWithoutInfinity)
-    {
-        static assert(hasSlicing!Range);
-        static assert(hasLength!Range);
-
-        Range r;
-        r.arr = 10.iota.array; // for clarity
-
-        assert(r.slide!(No.withFewerElements)(6).equal!equal(
-            [[0, 1, 2, 3, 4, 5], [1, 2, 3, 4, 5, 6], [2, 3, 4, 5, 6, 7],
-            [3, 4, 5, 6, 7, 8], [4, 5, 6, 7, 8, 9]]
-        ));
-        assert(r.slide!(No.withFewerElements)(16).empty);
-
-        assert(r.slide(4)[0 .. $].equal(r.slide(4)));
-        assert(r.slide(2)[$/2 .. $].equal!equal([[4, 5], [5, 6], [6, 7], [7, 8], [8, 9]]));
-        assert(r.slide(2)[$ .. $].empty);
-
-        assert(r.slide(3).retro.equal!equal(
-            [[7, 8, 9], [6, 7, 8], [5, 6, 7], [4, 5, 6], [3, 4, 5], [2, 3, 4], [1, 2, 3], [0, 1, 2]]
-        ));
-    }
-
-    // separate checks for infinity
-    auto infIndex = SliceableRange!(T, No.withOpDollar, Yes.withInfiniteness)([0, 1, 2, 3]);
-    assert(infIndex.slide(2)[0].equal([0, 1]));
-    assert(infIndex.slide(2)[1].equal([1, 2]));
-
-    auto infDollar = SliceableRange!(T, Yes.withOpDollar, Yes.withInfiniteness)();
-    assert(infDollar.slide(2)[1 .. $].front.equal([1, 2]));
-    assert(infDollar.slide(4)[0 .. $].front.equal([0, 1, 2, 3]));
-    assert(infDollar.slide(4)[2 .. $].front.equal([2, 3, 4, 5]));
+    static foreach (Partial; [Yes.withPartial, No.withPartial])
+    {{
+        static foreach (Range; SliceableDummyRanges)
+        {{
+            Range r;
+            r.arr = 10.iota.array; // use a 0-based array (for clarity)
+
+            static assert(isForwardRange!Range);
+            enum hasSliceToEnd = hasSlicing!Range && is(typeof(Range.init[0 .. $]) == Range);
+
+            assert(r.slide!Partial(2)[0].equal([0, 1]));
+            assert(r.slide!Partial(2)[1].equal([1, 2]));
+
+            // saveable
+            auto s = r.slide!Partial(2);
+            assert(s[0 .. 2].equal!equal([[0, 1], [1, 2]]));
+            s.save.popFront;
+            assert(s[0 .. 2].equal!equal([[0, 1], [1, 2]]));
+
+            assert(r.slide!Partial(3)[1 .. 3].equal!equal([[1, 2, 3], [2, 3, 4]]));
+        }}
+
+        static foreach (Range; SliceableDummyRangesWithoutInfinity)
+        {{
+            static assert(hasSlicing!Range);
+            static assert(hasLength!Range);
```

Comment:
Again, these aren't really necessary when `SliceableDummyRangesWithoutInfinity` is user defined